tests/integration/targets/aci_l3out_bfd_interface_profile/tasks/main.yml

```diff
@@ -0,0 +1,174 @@
+
+  - name: Add a new L3Out BFD Interface Profile - normal mode
+    cisco.aci.aci_l3out_bfd_interface_profile:
+      <<: *present_l3out_bfd_int_profile
+      state: present
+    register: nm_add_l3out_bfd_int_prof
+
+  - name: Verify adding L3Out BFD Interface Profile
+    assert:
+      that:
+      - cm_add_l3out_bfd_int_prof is changed
+      - nm_add_l3out_bfd_int_prof is changed
+      - cm_add_l3out_bfd_int_prof.previous == nm_add_l3out_bfd_int_prof.previous == []
+      - cm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.dn == nm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.dn == "uni/tn-ansible_tenant/out-ansible_l3out/lnodep-ansible_node_profile/lifp-ansible_interface_profile/bfdIfP"
+      - cm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.keyId == nm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.keyId == "3"
+      - cm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.type == nm_add_l3out_bfd_int_prof.proposed.bfdIfP.attributes.type == "none"
+
+  - name: Add L3Out BFD Interface Profile again- idempotency
+    cisco.aci.aci_l3out_bfd_interface_profile:
+      <<: *present_l3out_bfd_int_profile
+      state: present
+    register: add_l3out_bfd_int_prof_again
+
+  - name: Verify adding L3Out BFD Interface Profile again - idempotency
+    assert:
+      that:
+      - add_l3out_bfd_int_prof_again is not changed
+
+  # CREATE L3Out BFD Interface policy
+  - name: Add a new  L3Out BFD Interface policy
+    cisco.aci.aci_bfd_interface_policy:
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_bfd__interface_policy
+      description: Ansible BFD Interface Policy
+      state: present
+    register: add_bfd_interface_pol
+
+  # Update L3Out BFD Interface Profile
+  - name: Update L3Out BFD Interface Profile
+    cisco.aci.aci_l3out_bfd_interface_profile:
+      <<: *present_l3out_bfd_int_profile
+      interface_profile_type: sha1
+      key: "abc*123"
+      key_id: 15
+      bfd_interface_policy: ansible_bfd_interface_policy
+      state: present
+    register: update_l3out_bfd_int_prof
+
+  - name: Verify updating L3Out BFD Interface Profile
+    assert:
+      that:
+      - update_l3out_bfd_int_prof is changed
+      - update_l3out_bfd_int_prof.sent.bfdIfP.attributes.keyId == "15"
+      - update_l3out_bfd_int_prof.sent.bfdIfP.attributes.key == "abc*123"
+      - update_l3out_bfd_int_prof.sent.bfdIfP.attributes.type == "sha1"
+      - update_l3out_bfd_int_prof.sent.bfdIfP.children.0.bfdIfPol.attributes.tnBfdIfPolName == "ansible_bfd_interface_policy"
+
+  - name: Query L3Out BFD Interface Profile
+    cisco.aci.aci_l3out_bfd_interface_profile:
+      <<: *present_l3out_bfd_int_profile
+      state: query
+    register: query_l3out_bfd_int_prof
+
+  - name: Query all L3Out BFD Interface Profile
+    cisco.aci.aci_l3out_bfd_interface_profile:
+      <<: *aci_info
+      state: query
+    register: query_all_l3out_bfd_int_prof
+
+  - name: Delete L3Out BFD Interface Profile - check mode
```

Comment:
please add checkmode and idempotency for removal?

Reply:
added